core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -778,8 +766,9 @@ private boolean ignoreTimeout(ResponseFilter responseFilter) {
 
       Map<Address, Response> retval = new HashMap<>(CollectionFactory.computeCapacity(rsps.size()));
       boolean hasResponses = false;
+      Iterator<Map.Entry<Address, ReplicableCommand>> rpcCommandsIterator = rpcCommands.entrySet().iterator();
       for (Rsp<Response> rsp : rsps) {
```

Comment:
The rsps List seems redundant, we can just use the futures array directly and avoid an additional allocation and copy.